src/firefly/js/ui/UploadTableChooser.js

```diff
@@ -127,9 +127,37 @@ function existingTableSubmit(request,setUploadInfo) {
     return false;
 }
 
-function getSelectedColumns(columns) {
-    const {lonCol='', latCol=''} = findTableCenterColumns({tableData:{columns}}) ?? {}; //centerCols
-    const columnsSelected = columns.map((col) => col.name === lonCol || col.name === latCol? ({...col, use:true}) :  ({...col, use:false})); //select position cols only
+/*
+    if showUploadTableChooser is called with a defaultColsEnabled object, use the colTypes and colCount to set the selected columns
+    for this uploaded table
+ */
+function defaultColumnsSelector(columns,colTypes,colCount) {
+    let cols = [];
+    if  (columns == null) return null;
+    let count = 0;
+    for (let i=0; i< columns?.length; i++) {
+        if (count < colCount && colTypes.includes(columns[i].type)) {
+            cols.push(columns[i]);
+            count ++;
+        }
+    }
```

Comment:
I am not sure but this looks like a `filter` or a `filter` and a `map`. Using though will make she you are doing clearer.